applications/ChatGPT/chatgpt/trainer/callbacks/checkpoint.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+import torch.distributed as dist
+from chatgpt.trainer.strategies import ColossalAIStrategy
+from chatgpt.trainer.utils import is_rank_0
+
+from .base import Callback
+
+
+class SaveCheckpointCallback(Callback):
+    """
+        The callback for saving checkpoint.
+
+        It would take diffencent actions automaticly according to the training strategy.
+        The expected checkpoint files for different strategies:
+            Naive:
+                actor-model.pt
+                critic-model.pt
+                actor-optim-rank-0.pt
+                critic-optim-rank-0.pt
+            DDP:
+                Same as Naive strategy.
+            ColossalAIStrategy:
+                (Take 2 GPUs as an example)
+                actor-model.pt
+                critic-model.pt
+                actor-optim-rank-0.pt
+                actor-optim-rank-1.pt
+                critic-optim-rank-0.pt
+                critic-opimt-ranl-1.pt
+
+    Args:
+        path(str): The path you want to save checkpoint.
+        interval(int): The interval episode of saving checkpoint.
+
+    """
+
+    def __init__(
+        self,
+        path: str,
+        interval: int,
```

Comment:
Can you add args to control which model/optimizer to save?

Reply:
Done